### PR TITLE
fix UnaryFactor jacobian

### PR DESCRIPTION
**Pull request by nmahabadi**
* Fix Unaryfactor jacobian in LocalizationExample.cpp.
* Update gtsam document.

### CODE REVIEW

**Comment by varunagrawal** [comment]
@nmahabadi can you please pull the latest `develop` and merge it into this PR?

**Comment by nmahabadi** [comment]
Hi @varunagrawal, I fetched the latest from develop and rebased this branch to it. 
Let me know if that's okey.

**Comment by varunagrawal** [comment]
Would have preferred a merge to avoid potential conflicts, but this works. :slightly_smiling_face:

File: doc/gtsam.lyx [hunk 1]
```diff
@@ -1770,7 +1772,7 @@ filename "Code/LocalizationOutput5.txt"
 Comparing this with the covariance matrices in Section 
 \begin_inset CommandInset ref
 LatexCommand ref
-reference "sub:Full-Posterior-Inference"
+reference "subsec:Full-Posterior-Inference"
```
Comment (varunagrawal) [inline doc/gtsam.lyx:1775]: Looks good

**Review by varunagrawal — APPROVED** [review]
LGTM

@dellaert please take a look when you can.

**Review by dellaert — CHANGES_REQUESTED** [review]
Many unrelated changes in build files DO NOT MERGE

**Comment by dellaert** [comment]
I think something went wrong with rebase :-)
Typically, we ask people to *merge* develop into published branches rather than rebase to avoid surprises like this. Could you open a new PR starting from develop, cherry pick your relevant changes?

**Comment by nmahabadi** [comment]
Ok I will close this PR and open a new one.

**Pull request closed without merging**